nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -616,12 +616,16 @@ public void cancel() {
     }
 
     private DocumentModel createDocumentModelFromTypeName(String typeName, Map<String, Serializable> options) {
-        SchemaManager schemaManager = Framework.getService(SchemaManager.class);
-        DocumentType docType = schemaManager.getDocumentType(typeName);
+        return createDocumentModelFromParentAndType(null, typeName, options);
+    }
+
+    private DocumentModel createDocumentModelFromParentAndType(DocumentRef parentRef, String typeName,
```

Comment:
Don't we want to use this new API with `parentRef` for former methods with `String`? I'm talking about the `setPathInfo` in this method which could be removed I think.

Reply:
I see what you mean, but perhaps we should keep it, because this method compute the `path` and the `ref`:
```
 @Override
    public void setPathInfo(String parentPath, String name) {
        path = new Path(parentPath == null ? name : parentPath + '/' + name);
        ref = new PathRef(parentPath, name);
    }
```
i think if we want to remove it from former method, we should rewrite this once, and I thinking about the case where the `DocumentRef parentRef` that we pass as parameter if it is a `IdRef` we can do a meeting to clarify the things :)

Reply:
Of course for a meeting 👌

Some answer:
Idref shouldn't be a problem as we resolve the reference to get the path from Document.